backend/asmgen.ml

```diff
@@ -454,9 +527,9 @@ let linear_gen_implementation filename =
     | Func f -> emit_fundecl f
   in
   start_from_emit := true;
-  emit_begin_assembly ();
+  emit_begin_assembly (fun () -> ());
```

Comment:
This is used for FDO. It needs to do the same as the normal compilation, or files generated by FDO won't have dwarf info.

Reply:
`linear_gen_implementation` is now using `emit_begin_assembly_with_dwarf`